src/dust/dust.cpp

```diff
@@ -227,6 +285,81 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   return dust;
 }
 
+//----------------------------------------------------------------------------------------
+//! \fn  TaskStatus Dust::UpdateDustStoppingTime
+//  \brief Wrapper function for update dust stopping time
+template <Coordinates GEOM>
+TaskStatus UpdateDustStoppingTime(MeshData<Real> *md) {
+
+  using parthenon::MakePackDescriptor;
+
+  auto pm = md->GetParentPointer();
+
+  auto &dust_pkg = pm->packages.Get("dust");
+  if ((!dust_pkg->template Param<bool>("enable_coagulation"))) {
+    return TaskStatus::complete;
+  }
+
+  auto &resolved_pkgs = pm->resolved_packages;
+
+  IndexRange ib = md->GetBoundsI(IndexDomain::interior);
+  IndexRange jb = md->GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = md->GetBoundsK(IndexDomain::interior);
+
+  const int nspecies = dust_pkg->template Param<int>("nspecies");
```

Comment:
this will not be compatible with sparsity.  In general, I am not seeing any accesses via `sparse_id`.  How should we proceed on this front @brryan and @adamdempsey90?

Reply:
I think let's defer sparsity to a subsequent effort.

Reply:
I don't think it will be hard to at least use `sparse_id` when you are accessing properties.